### ISSUE TITLE
Issue with Type[None] and Type[NoneType]

### ISSUE THREAD

**Issue by bijij** [microsoft/pyright#2927]
**Describe the bug**
I was in a position where a function I had, say `foo` would expect either a `type` of `BaseException` or the `NoneType` type itself. (i.e. `types.NoneType`) originally I had it annotated as

```py
def foo(type: type[BaseException] | type[None]) -> None:
    ...
```

And this works fine for say `foo(type(None))`
However `foo(type(None.__class__))` won't type-check

**To Reproduce**
calling `reveal_type` as such:
```py
reveal_type(None)
reveal_type(type(None))
reveal_type(None.__class__)
```
we get the following:
```
test.py:1:13 - information: Type of "None" is "None"
test.py:2:13 - information: Type of "type(None)" is "Type[None]"
test.py:3:13 - information: Type of "None.__class__" is "Type[NoneType]
```

**Expected behavior**
At runtime type(None) and None.__class__ are both the class `NoneType`.
I would have expected that `Type[None]` and `Type[NoneType]` be treated as equivalent.
Or perhaps both should be revealed to be `Type[None]`?

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/8677174/150719599-b85cf198-de30-4fe6-9904-6d30a206c898.png)

**VS Code extension or command-line**
pyright 1.1.214

**Additional context**
mypy will output the following:
```
test.py:1: note: Revealed type is "None"
test.py:2: note: Revealed type is "Type[None]"
test.py:3: note: Revealed type is "Type[None]
```

would this behaviour be more desirable for pyright?

**Comment by erictraut** [comment]
I agree that `foo(None.__class__)` should type check. The types `Type[None]` and `Type[NoneType]` should be equivalent, so `None.__class__` should be assignable to a parameter of type `Type[None]`. I've introduced a fix that will make this work. This fix will be in the next release.

I don't think that `foo(type(None.__class__))` should type check in this case. I presume that was a typo in your bug report.

The runtime type definition for `NoneType` was private until Python 3.8. As of 3.8, the symbol was publicly documented and exposed. It is conditionally defined in the `types.pyi` stub. Pyright uses this type if it is present in the stub. That's where the type definition of `None.__class__` comes from.

**Comment by bijij** [comment]
> I don't think that foo(type(None.__class__)) should type check in this case. I presume that was a typo in your bug report.

You're correct that was a typo, my bad there.

**Comment by erictraut** [comment]
This is included in pyright 1.1.215, which I just published. It will also be included in the next release of pylance.
